device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java

```diff
@@ -331,270 +315,10 @@ public void completeFileUpload(FileUploadCompletionNotification notification) th
         this.fileUpload.sendNotification(notification);
     }
 
-    /**
-     * Retrieves the twin's latest desired properties
-     * @throws IOException if the iothub cannot be reached
-     */
-    public void getDeviceTwin() throws IOException
```

Comment:
All of these twin and method APIs that are shared between DeviceClient and ModuleClient have been moved into their parent class InternalClient for simplicity. In this move, though, many methods were renamed by appending an "Async" suffix to them.

Reply:
What confuses me is that we are moving public APIs that customers call into "InternalClient". But the word Internal throws me off because now it's been accessed by external entities, such as customer code.

Reply:
This is more of a comment than a suggestion to change any code :)

Reply:
While this is a bit confusing for us to read, the good news is that customers won't see this level of detail. They'll just see that both ModuleClient and DeviceClient have APIs like sendEventAsync()

Reply:
We could certainly rename this client to something like "CommonClient" or something to make it clear that the purpose of this client is not to be internal, but to house the shared APIs